code/modules/ai/action_states/action_state.dm

```diff
@@ -0,0 +1,57 @@
+//A singleton action state; AI components are added to the list and a subsystem will periodically call the process() for this
+/datum/element/action_state
+
+/datum/element/action_state/New()
+	..()
+	if(src)
+		SSactionstate.processing += src
+
+/datum/element/action_state/Attach(mob/living/carbon/mob)
+	if(mob && istype(mob))
+		..()
```

Comment:
Use `. = ..()` at the top of the proc unless you have a good reason otherwise

Reply:
Mob check is there to prevent adding action check to things like obj as they don't use the move_delay like carbons, if I call parent Attach() it's going to register a signal for COMSIG_PARENT_QDELETING but if the target isn't a mob then it's just a unnecessary signal registered

Reply:
ELEMENT_INCOMPATIBLE happening is a runtime error and if it happens you have other problems than some side effects, let those side effects happen

Reply:
Removed return ELEMENT_INCOMPATIBLE